src/main/groovy/org/jenkinsci/gradle/plugins/jpi/LicenseTask.groovy

```diff
@@ -7,28 +7,20 @@ import org.gradle.api.artifacts.Configuration
 import org.gradle.api.artifacts.Dependency
 import org.gradle.api.artifacts.ModuleVersionIdentifier
 import org.gradle.api.artifacts.ResolvedArtifact
-import org.gradle.api.artifacts.component.ProjectComponentIdentifier
-import org.gradle.api.file.FileCollection
-import org.gradle.api.tasks.InputFiles
-import org.gradle.api.tasks.Internal
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Classpath
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 
 class LicenseTask extends DefaultTask {
-    @Internal
-    Set<Configuration> configurations
 
-    @Internal
-    Set<Configuration> providedConfigurations
+    @Classpath
+    final Property<Configuration> libraryConfiguration = project.objects.property(Configuration)
```

Comment:
Does this work with a classpath input? I thought we don't want to support lazy types (aka `FileCollection` within lazy types (`Property`). What is the file collection which is used as input here?

Reply:
After more rework this became a plain `Configuration`. I changed it to work without providers/properties.